File: src/tree/lc100.java
package tree;

import util.TreeNode;

public class lc100 {
	public boolean isSameTree(TreeNode p, TreeNode q) {
		if (p == null && q == null) {
			return true;
		} else if (p != null && q != null) {
			return isSameTree(p.left, q.left) && isSameTree(p.right, q.right) && p.val == q.val;
		}
		return false;
	}
}

File: c++/tree/lc100.cpp

class Solution {
	public:
		bool isBalanced(TreeNode* root) {
			if(root==nullptr) return true;

			return abs(height(root->left) - height(root->right)) <= 1 && isBalanced(root->left) && isBalanced(root->right);

		}


		int height(TreeNode* root) {
			if(root==nullptr) return 0;
			return max(height(root->left),height(root->right))+1;
		}
};
